src/transformers/utils/generic.py

```diff
@@ -774,7 +774,7 @@ def torch_float(x):
 
     import torch
 
-    return x.to(torch.float32) if torch.jit.is_tracing() else int(x)
+    return x.to(torch.float32) if torch.jit.is_tracing() and isinstance(x, torch.Tensor) else int(x)
```

Comment:
And here torch_float

Reply:
I think since we're explicitly calling `torch_float` where float was used to be it should be fine, no?